aptos-move/framework/aptos-framework/sources/native_bridge.move

```diff
@@ -0,0 +1,794 @@
+    use aptos_std::smart_table;
+    use aptos_std::smart_table::SmartTable;
+    use aptos_framework::ethereum::EthereumAddress;
+    use aptos_framework::system_addresses;
+    use std::signer;
+
+    friend aptos_framework::native_bridge;
+
+    #[test_only]
+    use std::hash::sha3_256;
+    #[test_only]
+    use aptos_framework::ethereum;
+    #[test_only]
+    use aptos_framework::native_bridge_configuration;
+
+    /// Error codes
+    const EINVALID_PRE_IMAGE : u64 = 0x1;
+    const ENONCE_NOT_FOUND : u64 = 0x2;
+    const EZERO_AMOUNT : u64 = 0x3;
+    const EINVALID_BRIDGE_TRANSFER_ID : u64 = 0x4;
+    const ENATIVE_BRIDGE_NOT_ENABLED : u64 = 0x5;
+    const EINCORRECT_NONCE : u64 = 0x6;
+    const EID_NOT_FOUND : u64 = 0x7;
+
+    const MAX_U64 : u64 = 0xFFFFFFFFFFFFFFFF;
+
+    struct AddressPair<Initiator: store, Recipient: store> has store, copy {
+        initiator: Initiator,
+        recipient: Recipient,
+    }
+
+    /// A smart table wrapper
+    struct SmartTableWrapper<K, V> has key, store {
+        inner: SmartTable<K, V>,
+    }
+
+    /// Details on the outbound transfer
+    struct OutboundTransfer has store, copy {
+        bridge_transfer_id: vector<u8>,
+        initiator: address,
+        recipient: EthereumAddress,
+        amount: u64,
+    }
+
+    /// Checks if a bridge transfer ID is associated with an incoming nonce.
+    /// @param bridge_transfer_id The bridge transfer ID.
+    /// @return `true` if the ID is associated with an existing incoming nonce, `false` otherwise.
+    public(friend) fun is_incoming_nonce_set(bridge_transfer_id: vector<u8>): bool acquires SmartTableWrapper {
+        let table = borrow_global<SmartTableWrapper<vector<u8>, u64>>(@aptos_framework);
+        smart_table::contains(&table.inner, bridge_transfer_id)
+    }
+
+    /// Initializes the initiators tables and nonce.
+    ///
+    /// @param aptos_framework The signer for Aptos framework.
+    public fun initialize(aptos_framework: &signer) {
+        system_addresses::assert_aptos_framework(aptos_framework);
+
+        let nonces_to_details = SmartTableWrapper<u64, OutboundTransfer> {
+            inner: smart_table::new(),
+        };
+
+        move_to(aptos_framework, nonces_to_details);
+
+        let ids_to_incoming_nonces = SmartTableWrapper<vector<u8>, u64> {
+            inner: smart_table::new(),
+        };
+
+        move_to(aptos_framework, ids_to_incoming_nonces);
+    }
+
+    /// Creates bridge transfer details with validation.
+    ///
+    /// @param initiator The initiating party of the transfer.
+    /// @param recipient The receiving party of the transfer.
+    /// @param amount The amount to be transferred.
+    /// @param nonce The unique nonce for the transfer.
+    /// @return A `BridgeTransferDetails` object.
+    /// @abort If the amount is zero or locks are invalid.
+    public(friend) fun create_details(initiator: address, recipient: EthereumAddress, amount: u64, nonce: u64)
```

Comment:
Why do we need to create the details?

Reply:
In `initiate_bridge_transfer`, `native_bridge_store::create_details` is called, to create an `OutboundTransfer` struct with the details of the transfer. Then the details are stored in a smart table with `native_bridge_store::add`. 

This is needed so that the relayer can check the table for pending transfers and compare them with their mapping of bridge transfer IDs to incoming nonces on the destination chain.